src/dev/jest/config.js

```diff
@@ -93,7 +93,7 @@ export default {
   transformIgnorePatterns: [
     // ignore all node_modules except monaco-editor which requires babel transforms to handle dynamic import()
     // since ESM modules are not natively supported in Jest yet (https://github.com/facebook/jest/issues/4842)
-    '[/\\\\]node_modules(?![\\/\\\\](monaco-editor|weak-lru-cache|ordered-binary))[/\\\\].+\\.js$',
+    '[/\\\\]node_modules(?![\\/\\\\](monaco-editor|weak-lru-cache|ordered-binary|d3-color))[/\\\\].+\\.js$',
```

Comment:
This comment is outdated - it explains why `monaco-editor` is ignored, but not the other packages. What's the reason for needing to also ignore `d3-color`?

Reply:
there will be another 30+ unit tests fails. I pasted one below. and I think the reason is similar. my understanding is that `d3-color` is pre-compiled source code and it also needs babel transforms.


```
 FAIL  src/plugins/vis_type_timeseries/public/application/visualizations/views/timeseries/decorators/bar_decorator.test.js
  ● Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /home/ubuntu/work/OpenSearch-Dashboards/node_modules/d3-color/src/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){export {default as color, rgb, hsl} from "./color.js";
                                                                                      ^^^^^^

    SyntaxError: Unexpected token 'export'

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1728:14)
      at Object.<anonymous> (node_modules/@elastic/charts/src/common/color_library_wrappers.ts:21:1)
```

Reply:
Got it, and it wasn't an issue before because we weren't taking a direct dependency on it. Do you mind updating the comment to be more generic? Something like 

```
// ignore all node_modules except those which require babel transforms to handle dynamic import()
```

Reply:
yeah I definitely should
thanks for reminding me